src/Miunie.WindowsApp/ViewModels/StatusPageViewModel.cs

```diff
@@ -106,28 +109,36 @@ private async void CheckForTokenInClipboard()
         {
             if (!string.IsNullOrWhiteSpace(_miunie.BotConfiguration.DiscordToken)) { return; }
 
-            var clipboardContent = Clipboard.GetContent();
+            try
+            {
+                var clipboardContent = Clipboard.GetContent();
+
+                if (!clipboardContent.AvailableFormats.Contains(StandardDataFormats.Text)) { return; }
 
-            if (!clipboardContent.AvailableFormats.Contains(StandardDataFormats.Text)) { return; }
+                var possibleToken = await Clipboard.GetContent().GetTextAsync();
 
-            var possibleToken = await Clipboard.GetContent().GetTextAsync();
+                if (!_tokenManager.StringHasValidTokenStructure(possibleToken)) { return; }
+
+                var clipboardTokenDialog = new ContentDialog
+                {
+                    Title = "Paste copied bot token?",
+                    Content = "It looks like you have a bot token copied.\nDo you want to use it?",
+                    PrimaryButtonText = "Sure",
+                    CloseButtonText = "No, thanks"
+                };
 
-            if (!_tokenManager.StringHasValidTokenStructure(possibleToken)) { return; }
+                var result = await clipboardTokenDialog.ShowAsync();
 
-            var clipboardTokenDialog = new ContentDialog
+                if (result == ContentDialogResult.Primary)
+                {
+                    _tokenManager.ApplyToken(possibleToken, _miunie);
+                    RaisePropertyChanged(nameof(SettingsButtonIsVisable));
+                    RaisePropertyChanged(nameof(ActionCommand));
+                }
+            }
+            catch (Exception ex)
             {
-                Title = "Paste copied bot token?",
-                Content = "It looks like you have a bot token copied.\nDo you want to use it?",
-                PrimaryButtonText = "Sure",
-                CloseButtonText = "No, thanks"
-            };
-
-            var result = await clipboardTokenDialog.ShowAsync();
-
-            if (result == ContentDialogResult.Primary) { 
-                _tokenManager.ApplyToken(possibleToken, _miunie);
-                RaisePropertyChanged(nameof(SettingsButtonIsVisable));
-                RaisePropertyChanged(nameof(ActionCommand));
+                _logWriter.LogError($"Unable to query clipboard: {ex.Message}");
             }
         }
```

Comment:
_(optional)_
I feel like this try/catch statement is getting a bit complex.
I'd suggest extracting it into `TryApplyTokenFromClipboard()` at the very least. 😊